cpp/oneapi/dal/backend/primitives/stat/test/cov_dpc.cpp

```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/test/engine/common.hpp"
+#include "oneapi/dal/test/engine/fixtures.hpp"
+#include "oneapi/dal/test/engine/dataframe.hpp"
+#include "oneapi/dal/test/engine/math.hpp"
+#include "oneapi/dal/backend/primitives/stat/cov.hpp"
+
+namespace oneapi::dal::backend::primitives::test {
+
+namespace te = dal::test::engine;
+
+template <typename Float>
+class cov_test : public te::policy_fixture {
+public:
+    void check_correlation_for_uncorrelated_data(const ndarray<Float, 2>& corr) {
+        const auto corr_mat =
+            te::linalg::matrix<Float>::wrap(corr.get_data(),
+                                            { corr.get_dimension(0), corr.get_dimension(1) });
+
+        te::linalg::enumerate(corr_mat, [&](std::int64_t i, std::int64_t j, Float x) {
+            if (i == j) {
+                if (int(x) != 1) {
+                    CAPTURE(i, j, x);
+                    FAIL("Unexpected diagonal element of correlation matrix");
+                }
+            }
+            else {
+                if (int(x) != 0) {
+                    CAPTURE(i, j, x);
+                    FAIL("Unexpected non-diagonal element of correlation matrix");
+                }
+            }
+        });
+    }
+
+    void check_constant_variance(const ndarray<Float, 1>& vars, double expected_var) {
+        const auto vars_mat =
+            te::linalg::matrix<Float>::wrap(vars.get_data(), { vars.get_count(), 1 });
+
+        const double eps = te::get_tolerance<Float>(1e-6, 1e-12);
+
+        te::linalg::enumerate_linear(vars_mat, [&](std::int64_t i, Float var) {
+            if (std::abs(double(var) - expected_var) >= eps) {
```

Comment:
Is it for normalized data? If not, it might be better to use relative tolerance.